src/drivers/imx/edma.c

```diff
@@ -346,16 +346,9 @@ static int edma_setup_tcd(struct dma_chan_data *channel, int16_t soff,
 static int edma_set_config(struct dma_chan_data *channel,
 			   struct dma_sg_config *config)
 {
-	int handshake, irq;
 	int16_t soff = 0;
 	int16_t doff = 0;
 
-	/* We may need to pass some data through the handshake in the
-	 * future, so we just have them here, even if unused for now
-	 */
-	(void)handshake;
-	(void)irq;
-
 	tr_info(&edma_tr, "EDMA: set config");
```

Comment:
Have you considered `#if 0` for this?

Reply:
checkpatch gets mad at that, tried it for the `pdm_controllers_configuration` removal

Reply:
@marc-hb @cujomalainey I'm fine with this change. There is git history to keep this.

Reply:
You can't find in git history what you don't even know exists, that's why it's _sometimes_ worth keeping some dead code.  In fact it's not straight forward to find deleted code in git history even when you know it was deleted. Emphasis on _sometimes_ because I do NOT know this particular code, so maybe this one is NOT worth keeping despite the comment saying it should be kept.

As documented by the checkpatch maintainer, checkpatch is not always right and unlike cppcheck it's incremental so easy to ignore from time to time.

Reply:
@marc-hb In this specific case I'm fine with dropping the code despite my original comment. Something changed in handshake handling in the ESAI and SAI drivers and this code was indeed stale. Removing was the correct action.